sdk/core/azure-core-amqp/inc/azure/core/amqp/internal/link.hpp

```diff
@@ -70,7 +70,7 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
   class LinkEvents {
   public:
     virtual Models::AmqpValue OnTransferReceived(
-#if defined(TESTING_BUILD)
+#if defined(_azure_TESTING_BUILD)
         Link const& link,
 #else
         std::shared_ptr<LinkImpl> link,
```

Comment:
@LarryOsterman, is this really necessary? Can we have something here that works for both test and non-test builds?

Note: I am not suggesting that this is dangerous in the same way conditional virtual was. Here, the worst case, is that you get the unresolved linker symbol; but if you managed to compile and link, you can be sure that the app will work correctly.

Reply:
I can fix this but it significantly complicates the API surface internally.

The "Link" construct is a test hook, because the "Link" construct isn't exposed. All the internal code uses the LinkImpl class.